app/components/polaris/tag_component.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Polaris
+  class TagComponent < Polaris::NewComponent
+    renders_one :remove_button, -> (**system_arguments) do
+      Polaris::ButtonComponent.new(
+        classes: "Polaris-Tag__Button",
+        disabled: @disabled,
+        **system_arguments
+      ) do |button|
+        button.icon(name: "CancelSmallMinor")
+      end
+    end
+
+    def initialize(disabled: false, **system_arguments)
+      @disabled = disabled
+
+      @system_arguments = system_arguments
+      @system_arguments[:tag] = "span"
+      @system_arguments[:classes] = class_names(
+        @system_arguments[:classes],
+        "Polaris-Tag",
+        "Polaris-Tag--disabled" => disabled,
+        "Polaris-Tag--removable" => remove_button.present?
```

Comment:
@forsbergplustwo the issue is in that line. You trying to call `remove_button` slot here but it's not accessible in the initializer. We can call it only during rendering. And `view_component` doesn't provide any error in that case. Simply fails to render content. I also spent some time trying to figure it out the first time 😅

The solution is simple - extract `system_arguments` into a separate method and call it inside view. Example: 
https://github.com/baoagency/polaris_view_components/blob/main/app/components/polaris/text_field_component.rb#L103

Reply:
Maybe we should use this approach everywhere by default to avoid confusion in the future.

Reply:
Thanks so much @kirillplatonov , that makes sense 😊 Will update the PR and get back to you once fixed! And yes, I’ve noticed View Component swallow a lot of errors, so anything that makes it more foolproof the better it will be 👍